<a href="https://colab.research.google.com/github/ky3ob/dz_Kuznecov/blob/main/%D0%AD%D0%BA%D0%B7%D0%B0%D0%BC%D0%B5%D0%BD_%D0%9A%D1%83%D0%B7%D0%BD%D0%B5%D1%86%D0%BE%D0%B2%20%D0%A0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [14]:
expected = [50,50,50,50,50]
observed = [67, 50, 47, 53, 60]

import scipy.stats as stats

#Тест соответствия Хи-квадрат
stats.chisquare(f_obs=observed, f_exp=expected)



ValueError: For each axis slice, the sum of the observed frequencies must agree with the sum of the expected frequencies to a relative tolerance of 1e-08, but the percent differences are:
0.108

Код выдает ошибку, из-за большого значения в поле 1 (67), коэфициент равен p= 0.108 => переменная не подчинаяется предпологаемому распределению и значит что распределения статистически не схожи.

In [10]:
import numpy as np

n = 10000
time_between_calls = np.random.default_rng().exponential(scale=10, size=n)

x = ((time_between_calls < 5).sum())/n
y = ((time_between_calls < 4).sum())/n
x-y

0.06480000000000002

Вероятность на звонок в ближайшие 4-5 минут равна 0.065